src/test/llmnode_test.cpp

```diff
@@ -87,6 +96,74 @@ class LLMFlowHttpTest : public ::testing::Test {
         }
+
+        try {
+            plugin_config_t tokenizerPluginConfig = {};
+            std::string device = "CPU";
+            ov::genai::SchedulerConfig schedulerConfig = {
+                .max_num_batched_tokens = 256,
+                .cache_size = 8,
+                .block_size = 32,
+                .dynamic_split_fuse = true,
+                .max_num_seqs = 256,
+            };
+            plugin_config_t pluginConfig;
+            JsonParser::parsePluginConfig("", pluginConfig);
+            cbPipe = std::make_shared<ov::genai::ContinuousBatchingPipeline>("/ovms/llm_testing/facebook/opt-125m", schedulerConfig, device, pluginConfig, tokenizerPluginConfig);
+            llmExecutorWrapper = std::make_shared<LLMExecutorWrapper>(cbPipe);
+        } catch (const std::exception& e) {
+            SPDLOG_ERROR("Error during llm node initialization for models_path exception: {}", e.what());
+        } catch (...) {
+            SPDLOG_ERROR("Error during llm node initialization for models_path");
+        }
+    }
+
+    int generateExpectedText(std::string prompt) {
+        try {
+            auto generationHandle = cbPipe->add_request(
+                currentRequestId++,
+                prompt,
+                config);
+            if (generationHandle == nullptr) {
+                return -1;
+            }
+            llmExecutorWrapper->notifyNewRequestArrived();
+            std::vector<ov::genai::GenerationOutput> generationOutput = generationHandle->read_all();
+            std::sort(generationOutput.begin(), generationOutput.end(), [=](ov::genai::GenerationOutput& r1, ov::genai::GenerationOutput& r2) {
+                return r1.score > r2.score;
+            });
+            size_t i = 0;
+            for (ov::genai::GenerationOutput& out : generationOutput) {
+                if (i >= config.num_return_sequences)
+                    break;
+                i++;
+                std::vector<int64_t> tokens = out.generated_token_ids;
+                std::shared_ptr<ov::genai::Tokenizer> tokenizer = std::make_shared<ov::genai::Tokenizer>(cbPipe->get_tokenizer());
+                SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Generated tokens: {}", tokens);
+                std::string completion = tokenizer->decode(tokens);
+                expectedMessages.emplace_back(completion);
+            }
+        } catch (ov::AssertFailure& e) {
+            return -1;
+        } catch (...) {
+            return -1;
+        }
+        return 0;
+    }
+
+    int parsePartialReply(std::string response, rapidjson::Document& d) {
+        size_t pos = response.find("{");
+        if (pos == response.npos)
+            return -1;
+        response.erase(0, pos);
+        pos = response.find_last_of("}");
+        if (pos == response.npos)
+            return -1;
+        response.erase(pos + 1, response.size());
```

Comment:
Why do we need to do this?

Reply:
I think this code removes SSE prefix: `data: `, wouldn't it be better to just assert it exists and then remove first 5 characters?